### DISCUSSION TITLE
Ideas for the next release

### DISCUSSION THREAD

**Discussion by bolshakov** [bolshakov/stoplight#206] [Ideas]
Let's fantasise about what features are missing. Please use this thread to post your ideas.  

That's what I have in my mind:

1. Make it possible to display green lights in stoplight-admin 
2. Currently, Stoplight opens a circuit when X errors happen in a row. With the introduction of the sliding window, it makes more sense to open a circuit when X errors happen within this time window. 
3. It would be nice to see Stoplight's parameters in stoplight-admin
4. It would be nice te edit Stoplight's parameters in stoplight-admin

**Comment by ct-clearhaus** [comment]
Re. 1: A work-around is to first lock the light green, then unlock it again; that'll trigger the light's existence in the data store. We've used this strategy previously before when we had (short-lived) decentralized data stores to allow for a manual lock or colour change via stoplight-admin.

Re. 2: Agreed; or even when a specific rate of errors is surpassed (combined with some minimum). I've thought about taking a dive to produce a PR for that, but ... when things are on 🔥 I definitely want simplicity and things that are very easy to reason about, so I've always stopped myself even before starting. I haven't even spent the time researching what other CB libs in other languages allows, what experience folks have, etc. Do you have knowledge/experience/... to share in this regard?

Re. 3, 4: 👍

We've composed 2 lights to "fail-over from a primary to a secondary" (by turning the primary red and also allowing the secondary to go red). It'd be possible to do something more sophisticated, but I also fear the complexity in this 🤷‍♂️

**Comment by bolshakov** [comment]
One example that comes to my mind is Grafana's alert evaluation rules https://grafana.com/docs/grafana/latest/alerting/fundamentals/alert-rules/rule-evaluation/
